File: app/src/main/java/wangdaye/com/geometricweather/common/utils/NetworkUtils.java
package wangdaye.com.geometricweather.common.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtils {

    public static boolean isAvailable(Context context) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(
                Context.CONNECTIVITY_SERVICE);
        if (manager != null) {
            NetworkInfo activeNetworkInfo = manager.getActiveNetworkInfo();
            NetworkInfo vpnInfo = manager.getNetworkInfo(ConnectivityManager.TYPE_VPN);
            return activeNetworkInfo != null || (vpnInfo != null && vpnInfo.isConnected());
        }
        return false;
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/common/utils/NetworkUtils.kt
package wangdaye.com.geometricweather.common.utils

import android.content.Context
import android.net.ConnectivityManager

object NetworkUtils {

    @JvmStatic
    fun isAvailable(context: Context): Boolean {
        val manager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?
        if (manager != null) {
            @Suppress("DEPRECATION")
            return manager.activeNetworkInfo != null
        }
        return false
    }
}
